### PR TITLE
Supported ceph image's extending and shrinking

### PR DESCRIPTION
**Pull request by zhencliu**
1. Only ceph was supported
  2. Extended and shrinked ceph image

Signed-off-by: Zhenchao Liu <zhencliu@redhat.com>

ID: 1786240

### CODE REVIEW

**Comment by zhencliu** [comment]
@lolyu

**Comment by zhencliu** [comment]
@hellohellenmao 
--testcase=remote_block_resize

 ceph only

**Comment by hellohellenmao** [comment]
1 warning when applying the patch:
`# git am 2040.patch
Applying: Supported ceph image extending and shrinking
.git/rebase-apply/patch:41: trailing whitespace.
    
warning: 1 line adds whitespace errors.
`

**Comment by hellohellenmao** [comment]
Passed with drive.
`# python ConfigTest.py --testcase=remote_block_resize --machine=q35 --guestname=RHEL.8.1.0 --imageformat=raw --storage-backend-conf=/tmp/ceph.cfg --clone=no`

`(1/1) Host_RHEL.m8.u2.product_rhel.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.remote_block_resize.q35: PASS (112.98 s)
`

**Comment by hellohellenmao** [comment]
Passed with blockdev.
`# python ConfigTest.py --testcase=remote_block_resize --machine=q35 --guestname=RHEL.8.1.0 --imageformat=raw --storage-backend-conf=/tmp/ceph.cfg --clone=no --customsparams='qemu_force_use_drive_expression=no'`

`(1/1) Host_RHEL.m8.u2.product_rhel.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.remote_block_resize.q35: PASS (97.82 s)`

**Comment by zhencliu** [comment]
> 1 warning when applying the patch:
> `# git am 2040.patch
> Applying: Supported ceph image extending and shrinking
> .git/rebase-apply/patch:41: trailing whitespace.
> 
> warning: 1 line adds whitespace errors.`

fixed

**Comment by zhencliu** [comment]
without auth patch:
 (1/1) Host_RHEL.m8.u2.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.remote_block_resize: PASS (71.39 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 74.13 s

**Comment by zhencliu** [comment]
@luckyh @xutian  to be merged

File: qemu/tests/cfg/remote_block_resize.cfg [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Network storage backends:
+#   ceph
+# The following testing scenarios are covered:
+#   extend remote image
+#   shrink remote image
+# Note:
+#   Both -blockdev & -drive were supported
+
+- remote_block_resize:
```
Comment (luckyh) [inline qemu/tests/cfg/remote_block_resize.cfg]: `block_resize_with_remote_storage`

File: qemu/tests/cfg/remote_block_resize.cfg [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    only ceph
+    type = remote_block_resize
+    extend_ratio = 1.5
+    shrink_ratio = 0.9
+    disk_change_ratio = "${extend_ratio} ${shrink_ratio}"
+    accept_ratio = 0.005
+
+    qcow2:
+        # qcow2 is supported since 4.1.0 on ceph
+        required_qemu = [4.1.0,)
```
Comment (luckyh) [inline qemu/tests/cfg/remote_block_resize.cfg]: No need to have to this line.

File: qemu/tests/remote_block_resize.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        current_size = utils_disk.get_disk_size(session, os_type, disk)
+        accept_ratio = float(params.get("accept_ratio", 0))
+        if (current_size <= block_size and
+                current_size >= block_size * (1 - accept_ratio)):
+            logging.info("Block Resizing Finished !!! \n"
+                         "Current size %s is same as the expected %s",
+                         current_size, block_size)
+            return True
+        else:
+            logging.warn("Current: %s\nExpect: %s\n" % (current_size,
```
Comment (luckyh) [inline qemu/tests/remote_block_resize.py]: `logging.error`

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**